tests/linearizability/failpoints.go

```diff
@@ -234,6 +240,75 @@ func triggerCompact(ctx context.Context, member e2e.EtcdProcess) error {
 	return nil
 }
 
+// latestRevisionForEndpoint gets latest revision of the first endpoint in Client.Endpoints list
+func latestRevisionForEndpoint(ctx context.Context, c *clientv3.Client) (int64, error) {
+	cntx, cancel := context.WithTimeout(ctx, 100*time.Millisecond)
+	defer cancel()
+	resp, err := c.Status(cntx, c.Endpoints()[0])
+	if err != nil {
+		return 0, err
+	}
+	return resp.Header.Revision, err
+}
+
+func triggerBlackholeUntilSnapshot(t *testing.T, ctx context.Context, member e2e.EtcdProcess, clus *e2e.EtcdProcessCluster) error {
+	leader := clus.Procs[clus.WaitLeader(t)]
+	lc, err := clientv3.New(clientv3.Config{
+		Endpoints:            []string{leader.Config().ClientURL},
+		Logger:               zap.NewNop(),
+		DialKeepAliveTime:    1 * time.Millisecond,
+		DialKeepAliveTimeout: 5 * time.Millisecond,
+	})
+	if err != nil {
+		return err
+	}
+	defer lc.Close()
+
+	mc, err := clientv3.New(clientv3.Config{
+		Endpoints:            []string{member.Config().ClientURL},
+		Logger:               zap.NewNop(),
+		DialKeepAliveTime:    1 * time.Millisecond,
+		DialKeepAliveTimeout: 5 * time.Millisecond,
+	})
+	if err != nil {
+		t.Error(err)
+		return err
+	}
+	defer mc.Close()
+
+	proxy := member.PeerProxy()
+	// Blackholing will cause peers to not be able to use streamWriters registered with member
+	// but peer traffic is still possible because member has 'pipeline' with peers
+	// TODO: find a way to stop all traffic
+	proxy.BlackholeTx()
+	proxy.BlackholeRx()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+		}
+		// have to refresh revBlackholedMem because peer traffic is still getting through
+		// TODO: for unknown reason revBlackholedMem will stabilize at some point
```

Comment:
It's expected that revision might increase for some time on blackholed member. Revision increases only when change is applied to database. Changes are first negotiated via Raft, persisted to WAL and only then applied.

 I expect after blackholing member revision can still increase as it processes changes that it already received but not yet applied.

Reply:
Please remove the TODO